sqlmesh/core/engine_adapter/athena.py

```diff
@@ -373,6 +409,101 @@ def _find_matching_columns(
                 matches.append((key, match_dtype))
         return matches
 
+    def _insert_overwrite_by_time_partition(
+        self,
+        table_name: TableName,
+        source_queries: t.List[SourceQuery],
+        columns_to_types: t.Dict[str, exp.DataType],
+        where: exp.Condition,
+        **kwargs: t.Any,
+    ) -> None:
+        table_type = self._query_table_type(table_name)
+
+        if table_type == "iceberg":
+            # Iceberg tables work as expected, we can use the default behaviour
+            return super()._insert_overwrite_by_time_partition(
+                table_name, source_queries, columns_to_types, where, **kwargs
+            )
+
+        # For Hive tables, we need to drop all the partitions covered by the query and delete the data from S3
+        self._clear_partition_data(table_name, where)
```

Comment:
one shitty thing about this is we lose transactions, if something goes wrong here you lose data right

Reply:
That's right, but Athena doesn't do transactions anyway.

I think we can say that if you want atomic deletes then you'll need to use Iceberg